pkg/logs/launchers/file/launcher.go

```diff
@@ -118,6 +121,12 @@ func (s *Launcher) run() {
 // cleanup all tailers
 func (s *Launcher) cleanup() {
 	stopper := startstop.NewParallelStopper()
+	s.cleanUpRotatedTailers()
+	for _, tailer := range s.rotatedTailers {
+		stopper.Add(tailer)
+	}
+	s.rotatedTailers = []*tailer.Tailer{}
```

Comment:
I think this ends up called from a different goroutine than `cleanUpRotatedTailers` and `restartTailerAfterFileRotation` so probably needs to synchronize access to `rotatedTailers` and its contents.

Reply:
true - however the `stop` signal has finished first before `cleanup` is called which terminates the other goroutine. 

Maybe it would be cleaner/safer to move the `cleanup` call into the `case <-s.stop:` branch of the runloop to ensure it's always executed on the same goroutine?

Reply:
Indeed, the channel is sync, so writing to `s.stop` would block if the main loop is busy elsewhere.

> Maybe it would be cleaner/safer to move the cleanup call into the case <-s.stop: branch of the runloop to ensure it's always executed on the same goroutine?

I think it would be cleaner, but then `Launcher#Stop` would need to block on something else to wait for tailers to finish.